### PR TITLE
Updated settings dialog to download sources on save

### PR DESCRIPTION
**Pull request by RandomlyKnighted**

### CODE REVIEW

File: TemplatePack/SettingsForm.cs [hunk 1]
```diff
@@ -288,6 +286,15 @@ public void LoadSourcesListView(RemoteTemplateSettings rts)
             }
         }
 
+        private async void startRebuildingTemplates()
```
Comment (sayedihashimi) [inline TemplatePack/SettingsForm.cs]: ``async void``` should be avoided. More info at https://msdn.microsoft.com/en-us/magazine/jj991977.aspx. Change this to ```async Task```, and then in the code that calls it add ```await```.

**Comment by RandomlyKnighted** [comment]
@sayedihashimi how's that? Per the link you sent I left ```async void``` on the event handlers but changed everything else to use ```async Task```.

**Comment by sayedihashimi** [comment]
@RandomlyKnighted that looks good. Merged it. I made some further changes in 056f43370ee227bc55684fd023f06e6702ec959b to ensure that OK wasn't clicked more than once.

**Comment by RandomlyKnighted** [comment]
@sayedihashimi great! as far as I know that gets all the bugs fixed and should now make the article I sent you up to date

**Pull request merged**